model-server/build.gradle.kts

```diff
@@ -200,74 +211,60 @@ spotless {
-        globalProperties.putAll(
-            mapOf(
-                "models" to "",
-                "apis" to "",
-                "supportingFiles" to "Paths.kt",
-            ),
-        )
-    }
-
-    // Ensure that the OpenAPI generator runs before starting to compile
-    tasks.named("processResources") {
-        dependsOn(targetTaskName)
-    }
-    tasks.named("compileKotlin") {
-        dependsOn(targetTaskName)
-    }
+val openApiGenerationPath = project.layout.buildDirectory.get().dir("generated/openapi")
+val restApiPackage = "org.modelix.model.server.handlers"
+val openApiGenerate = tasks.register<GenerateTask>("openApiGenerateModelServer") {
+    dependsOn(openApiSpec)
+    // we let the Gradle OpenAPI generator plugin build data classes and API interfaces based on the provided
+    // OpenAPI specification. That way, the code is forced to stay in sync with the API specification.
+    generatorName.set("kotlin-server")
+    inputSpec.set(openApiSpec.resolve().first().toString())
+    outputDir.set(openApiGenerationPath.toString())
+    packageName.set(restApiPackage)
+    apiPackage.set(restApiPackage)
+    modelPackage.set(restApiPackage)
+    // We use patched mustache so that only the necessary parts (i.e. resources and models)
+    // are generated. additionally we patch the used serialization framework as the `ktor` plugin
+    // uses a different one than we do in the model-server. The templates are based on
+    // https://github.com/OpenAPITools/openapi-generator/tree/809b3331a95b3c3b7bcf025d16ae09dc0682cd69/modules/openapi-generator/src/main/resources/kotlin-server
+    templateDir.set("${layout.projectDirectory.dir("src/main/resources/openapi/templates")}")
+    configOptions.set(
+        mapOf(
+            // we use the ktor generator to generate server side resources and model (i.e. data classes)
+            "library" to "ktor",
+            // the generated artifacts are not built independently, thus no dedicated build files have to be generated
+            "omitGradleWrapper" to "true",
+            // the path to resource generation we need
+            "featureResources" to "true",
+            // disable features we do not use
+            "featureAutoHead" to "false",
+            "featureCompression" to "false",
+            "featureHSTS" to "false",
+            "featureMetrics" to "false",
+        ),
+    )
+    // generate only Paths and Models - only this set will produce the intended Paths.kt as well as the models
+    // the openapi generator is generally very picky and configuring it is rather complex
+    globalProperties.putAll(
+        mapOf(
+            "models" to "",
+            "apis" to "",
+            "supportingFiles" to "Paths.kt",
+        ),
+    )
+}
 
-    // add openAPI generated artifacts to the sourceSets
-    sourceSets["main"].kotlin.srcDir("$outputPath/src/main/kotlin")
+// Ensure that the OpenAPI generator runs before starting to compile
+tasks.named("processResources") {
```

Comment:
https://github.com/modelix/modelix.core/pull/353#discussion_r1413534794 this was the original reason. As we have dropped ktlint from Gradle, this might not be needed anymore. I'll check this.

Reply:
> Reason: Task ':model-server:compileKotlin' uses this output of task ':model-server:openApiGenerateModelServer' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.

Reply:
I could reduce this to what is still needed